libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -714,13 +717,22 @@ func `/`(a, b: Duration): float64 =
     fb = float64(b.nanoseconds)
   fa / fb
 
+proc disconnectPeer(peer: PubSubPeer) {.async.} =
+  if peer.sendConn != nil:
+    try:
+      await peer.sendConn.close()
```

Comment:
I think you want to use `switch.disconnect(peerId)` here, otherwise you're simply closing an mplex channel.

Reply:
thanks! I was gonna ask you indeed